apps/opentelemetry/src/otel_batch_olp.erl

```diff
@@ -0,0 +1,623 @@
+                                 resource=Resource,
+                                 atomic_ref=AtomicRef,
+                                 tables=Tabs,
+                                 external_config=ExtConfig,
+                                 otel_signal=Signal}) ->
+
+    NewCurrentTab = ?next_tab(CurrentTab),
+    %% the new table is expected to be empty or hold a few late writes from the previous export,
+    %% so it safe to set available max size
+    ?set_available(AtomicRef, NewCurrentTab, MaxSize),
+    ?set_current_tab(AtomicRef, NewCurrentTab),
+    export_async(Exporter, Resource, ?tab_name(CurrentTab, Tabs), ExtConfig, Signal).
+
+export_async(Exporter, Resource, CurrentTab, ExtConfig, Signal) ->
+    erlang:spawn_link(fun() -> export(Exporter, Resource, CurrentTab, ExtConfig, Signal) end).
+
+export(undefined, _, _, _, _) ->
+    true;
+export({ExporterModule, ExporterState}, Resource, Tab, ExtConfig, Signal) ->
+    try
+        %% TODO: API for both signals should probably be aligned in otel_exporter
+        TabOrTabConfig = case Signal of
+                             traces -> Tab;
+                             logs -> {Tab, ExtConfig}
+                         end,
+        %% we ignore values, as no retries mechanism, is implemented
+        otel_exporter:export(Signal, ExporterModule, TabOrTabConfig, Resource, ExporterState)
+    catch
+        Class:Reason:St ->
+            %% Other logger handler(s) (e.g. default) should be enabled, so that
+            %% log events produced by otel_log_handler are not lost in case otel_log_handler
+            %% is not functioning properly.
+            ?LOG_ERROR("~p exporter ~p failed with exception: ~p:~p, stacktrace: ~p",
+                       [Signal, Class, Reason, otel_utils:stack_without_args(St)]),
+            error
+    end.
+
+new_export_table(Name, ExtraEtsOpts) ->
+    %% log event timestamps used as keys are not guaranteed to always be unique,
+    %% so we use duplicate_bag
+    %% Using timestamps as keys instead of instrumentation scopes is expected
+    %% to have higher entropy which should improve write concurrency
+    Opts = lists:usort([public,
+                        named_table,
+                        {write_concurrency, true},
+                        duplicate_bag] ++ ExtraEtsOpts),
+    ets:new(Name, Opts).
+
+clear_table_and_disable(AtomicRef, Tabs) ->
+    case ?current_tab(AtomicRef) of
+        0 ->
+            %% already disabled
+            ok;
+        CurrentTab ->
+            ?disable(AtomicRef),
+            CurrentTabName = ?tab_name(CurrentTab, Tabs),
+            ets:delete_all_objects(CurrentTabName),
+            ok
+    end.
+
+complete_exporting(Data) ->
+    {next_state, idle, Data#data{runner_pid=undefined,
+                                 handed_off_table=undefined}}.
+
+kill_runner(Data=#data{runner_pid=RunnerPid, handed_off_table=Tab}) when RunnerPid =/= undefined ->
```

Comment:
Curious: why do we keep exporter linked at all?

Reply:
Thanks, I think it makes sense to monitor it instead. Linking is rather legacy.

Reply:
Changed to `spawn_monitor`